src/component/common/ProjectSelect/ProjectSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { MouseEventHandler, useMemo, VFC } from 'react';
+import { MenuItem, Typography } from '@material-ui/core';
+import DropdownMenu, { IDropdownMenuProps } from '../DropdownMenu/DropdownMenu';
+import useProjects from 'hooks/api/getters/useProjects/useProjects';
+import { IProjectCard } from 'interfaces/project';
+
+const ALL_PROJECTS = { id: '*', name: '> All projects' };
+
+interface IProjectSelectProps {
+    currentProjectId: string;
+    updateCurrentProject: (id: string) => void;
+}
+
+const ProjectSelect: VFC<IProjectSelectProps & Partial<IDropdownMenuProps>> = ({
+    currentProjectId,
+    updateCurrentProject,
+    ...rest
+}) => {
+    const { projects } = useProjects();
+
+    const setProject = (value?: string | null) => {
+        const id = value && typeof value === 'string' ? value.trim() : '*';
+        updateCurrentProject(id);
+    };
+
+    const currentProject = useMemo(() => {
+        const project = projects.find(i => i.id === currentProjectId);
+        return project || ALL_PROJECTS;
+    }, [currentProjectId, projects]);
+
+    const handleChangeProject: MouseEventHandler = event => {
```

Comment:
We should probably consider using something like `MouseEventHandler | TouchEventHandler` at some point. 

Something for later.